components/event/org.wso2.carbon.event.core/src/main/java/org/wso2/carbon/event/core/internal/builder/EventBrokerBuilder.java

```diff
@@ -74,8 +74,8 @@ public static EventBroker createEventBroker() throws EventBrokerConfigurationExc
      */
     private static OMElement loadConfigXML() throws EventBrokerConfigurationException {
 
-        String carbonHome = System.getProperty(ServerConstants.CARBON_HOME);
-        String path = carbonHome + File.separator + "repository" + File.separator + "conf" + File.separator + EventBrokerConstants.EB_CONF;
+        String carbonConfigHome = System.getProperty(ServerConstants.CARBON_CONFIG_DIR_PATH);
```

Comment:
In the previous implementation, the path was constructed. If the System property is not defined, then there is an issue.